jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/actions/ViewResourceAction.kt

```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.explorer.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.DumbAware
+import software.aws.toolkits.jetbrains.core.explorer.ViewResourceDialog
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerNode
+
+abstract class ViewResourceAction<in T : AwsExplorerNode<*>>(private val actionTitle: String, val resourceType: String, private val prefix: String) :
```

Comment:
IMO this is a bit awkward since it seems written specifically for the s3 protocol prefix. is there a cleaner way to make this work?

Reply:
Changed the prefix type to String with default as null so it shouldn't affect any other uses of this class. Another alternative I considered was changing the resourceType sent from ViewBucketAction as S3 Bucket Name/URI and edit that, but that would not be a great choice, but since now the prefix is nullable, it should be better?

Reply:
Added a function as a parameter instead of prefix to check different conditions for ensuring correct input text is entered